.github/workflows/behave.yml

```diff
@@ -17,63 +17,71 @@ jobs:
   behave:
     name: Run Behave
     runs-on: ubuntu-latest
-
-    services:
-      postgres:
-        image: postgres:latest
-        env:
-          POSTGRES_USER: scram
-          POSTGRES_PASSWORD: ''
-          POSTGRES_DB: test_scram
-          POSTGRES_HOST_AUTH_METHOD: trust
-        ports:
-          - 5432:5432
-        options: >-
-          --health-cmd "pg_isready -U scram"
-          --health-interval 10s
-          --health-timeout 5s
-          --health-retries 5
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: ['3.11', '3.12']
 
     steps:
       - name: Check out the code
         uses: actions/checkout@v4
 
       - name: Set up Docker
         uses: docker/setup-buildx-action@v3
+
       - name: Install Docker Compose
         run: |
           sudo apt-get update
           sudo apt-get install -y docker-compose make
 
+      - name: Check Docker state (pre-build)
+        run: docker ps
+
       - name: Build Docker images
         run: make build
+        env:
+          PYTHON_IMAGE_VER: "${{ matrix.python-version }}"
 
       - name: Migrate Database
         run: make migrate
 
       - name: Run Application
         run: make run
 
+      - name: Check Docker state (pre-test)
+        run: docker ps
+
       - name: Run pytest + behave with Coverage
```

Comment:
are we running pytest in behave as well?

Reply:
We need the full test suite to run somewhere so we get accurate coverage results, as I think pytest tests some code that behave doesn't, and vice-versa.

Reply:
Ah right, fair